File: java-demo/src/main/java/demo4/Ex1.java
package demo4;

import java.util.stream.IntStream;

import akka.actor.ActorSystem;
import akka.stream.ActorMaterializer;
import akka.stream.javadsl.Source;

public class Ex1 {

  public static void main(String[] args) {
    ActorSystem system = ActorSystem.create("demo4");
    ActorMaterializer materializer = ActorMaterializer.create(system);

    // Create a slow stream and show CPUs usage
    IntStream numbers = IntStream.iterate(1, x -> x + 1);
    Source<Integer, ?> source = Source.from(() -> numbers.iterator());
    source.map(e -> { Thread.sleep(1000); return e; }).runForeach(x -> System.out.println(x), materializer);
  }

}

File: scala-demo-seed/src/main/scala/demo4/Ex1.scala
package demo4

import akka.actor.ActorSystem
import akka.stream.ActorMaterializer
import akka.stream._
import akka.stream.scaladsl._

import scala.concurrent.Future
import scala.concurrent.Await
import scala.concurrent.duration.Duration
import scala.concurrent.duration._

object Ex1 extends App {
  implicit val system = ActorSystem("demo4")
  implicit val materializer = ActorMaterializer()
  import system.dispatcher

  // Create a slow stream and show CPUs usage

}